### DISCUSSION TITLE
mishards离线环境下启动失败

### DISCUSSION THREAD

**Discussion by tinygiants-tech** [milvus-io/milvus#6992] [Q&A and General discussion]
zilliz的技术工程师，你们号好，最近在搭建分布式milvus1.1.1服务器启动mishards集群分片中间件的时候遇到了一个问题，我按照官方Mishards示例要求，安装了python3.7.9, 下载了requirements.txt文件下的全部依赖，相关底层依赖也已经下载好，并且顺利安装，但是，当我使用python3 main.py启动mishards的时候报了错误，错误如下：
Traceback
File "~/mishards/settings.py", line 24, in <module>
    from utils.logger_helper import config
ModuleNotFoundError: No module named 'utils'

然后我尝试着在PyPI下载了utils-1.0.1，pip3安装后运行python3 main.py，再次报错
ModuleNotFoundError: No module named 'utils.logger_helper'

希望zilliz的同事能够帮我分析下这个依赖问题要如何解决，期待回复，谢谢！

**Comment by yhmo** [comment]
/mishards/settings.py 使用的utils是mishards自己的utils，在/mishards的同级目录 /utils
mishards的源码结构是这样的：https://github.com/milvus-io/milvus/tree/1.1/shards
shards目录下有mishards，tracer，utils，discovery这几个目录，都是需要的。
